fairseq/training.py

```diff
@@ -168,14 +208,13 @@ def __init__(
                 self.output_path("checkpoints/checkpoint{}.pt".format(k)),
                 k,
             )
-            for k in stored_epochs
-            if k in self.keep_epochs
+            for k in self.keep_epochs
         }
         self.out_cached_audio_manifest = self.output_path(
             "cached_audio_manifest", directory=True
         )
```

Comment:
I just noticed that this output will always exist, but the file is only written if `use_cache_manager` is True. This is not nice. could we do one of the following
1. only create the output if we use cache manager
```suggestion
        if use_cache_manager:
            self.out_cached_audio_manifest = self.output_path(
                "cached_audio_manifest", directory=True
            )
```
2. always create the file, but copy over the default manifest if we do not use the cache manager
3. never have this as an output but store it as a regular file in the work folder.